flink-formats/flink-orc/src/main/java/org/apache/flink/orc/OrcSplitReaderUtil.java

```diff
@@ -140,7 +140,12 @@ static TypeDescription logicalTypeToOrcType(LogicalType type) {
 			case CHAR:
 				return TypeDescription.createChar().withMaxLength(((CharType) type).getLength());
 			case VARCHAR:
-				return TypeDescription.createVarchar().withMaxLength(((VarCharType) type).getLength());
+				int len = ((VarCharType) type).getLength();
+				if (len == VarCharType.MAX_LENGTH) {
+					return TypeDescription.createString();
```

Comment:
What if user specifies `VARCHAR(2147483647)` in the schema and we convert it to string here? Will there be a problem?

Reply:
`VARCHAR(2147483647)` is `STRING` in Flink.
We don't need support real `VARCHAR(2147483647)` in orc for hive 2.0. Hive don't have this type.

Reply:
For newer versions, orc can schema evolution. So `VARCHAR(2147483647)` is supported.

Reply:
Got it. Thanks for the explanation.